### ISSUE TITLE
model.reload() don't update fragmentArray properties (using v6+ and ember-data 4.6)

### ISSUE THREAD

**Issue by ndekeister-us** [adopted-ember-addons/ember-data-model-fragments#467]
Hello!

First thanks a lot for the awesome work to make this repository compatible with Ember 4+ 🙏 

With version 6 of this addon and ember-data 4.6 we found a little issue, when using `model.reload()`, `fragmentArray` properties are not updated correctly

https://github.com/adopted-ember-addons/ember-data-model-fragments/assets/56396753/f18349d0-72d3-4e20-9520-1889b74784cf

If we are looking at `model.currentState.recordData.__fragmentData['myProperty']`, it contains correct values, but `model['myProperty']` still contains the initial value

Reproduced here -> https://github.com/ndekeister-us/ember-test-app_ember-data-model-fragments

**Comment by deanmarano** [comment]
I've also experienced not seeing reload on ED 3.28. It looks like [this test for reloading](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/f4f4daf8901f8dd4a6fef398eb3b0cebae8cf3b4/tests/integration/save_test.js#L713) is skipped, and it is failing if ran.

**Comment by deanmarano** [comment]
I checked out the sample repo here and [the PR that I got merged](https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/469) doesn't resolve this issue (because of [this rollback](https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/469/commits/45ad08c6ecd98ce3fbaf14eb27311e3cabad3659)). [This code](https://github.com/adopted-ember-addons/ember-data-model-fragments/blob/master/addon/record-data.js#L695-L697) is the issue (even though I thought it might not be necessary): 

```javascript
if ((updates[key] === null) !== (original[key] === null)) {
  changedKeys.push(key);
}
```
It handles if the fragment is null or is becoming null, but doesn't actually detect changes (as the `_changedFragmentKeys` function name implies).

I'm not sure what the best way to detect changes here. My initial attempt was checking if they `===` each other, which doesn't work for arrays (so it was always triggering, which _did_ fix the bug). I considered verifying on length and clientId:

```javascript
let wasOrBecameNull = original[key] === null || updates[key] === null;
let isDifferentLength = original[key]?.length !== updates[key].length;
let hasNewIds = !updates[key].every(record => original[key].includes(record.clientId));
if (wasOrBecameNull || isDifferentLength || hasNewIds) {
  changedKeys.push(key);
}
```
I've [tested this](https://github.com/adopted-ember-addons/ember-data-model-fragments/compare/master...deanmarano:ember-data-model-fragments:master) on the provided reproduction (thanks @ndekeister-us!) and it works (as well as a repro in my app), but I haven't written a test for it yet.

**Comment by deanmarano** [comment]
@ndekeister-us Can you verify that the latest commit (not released package) works for you?

**Comment by knownasilya** [comment]
Released as v6.0.2

**Comment by ndekeister-us** [comment]
Thanks a lot for the fix + release, tested on our main application and on the demo app and it is now working as expected 🎉
